generators/src/main/java/io/swagger/codegen/v3/generators/typescript/TypescriptGenerator.java

```diff
@@ -0,0 +1,72 @@
+package io.swagger.codegen.v3.generators.typescript;
+
+import io.swagger.codegen.v3.SupportingFile;
+import java.io.File;
+
+public class TypescriptGenerator extends AbstractTypeScriptClientCodegen {
+	protected String sourceFolder = "src";
+	protected String templateVersion = "0.0.1";
+
+	/**
+	 * Configures a friendly name for the generator.  This will be used by the generator
+	 * to select the library with the -l flag.
+	 *
+	 * @return the friendly name for the generator
+	 */
+	public String getName() {
+		return "typescript";
+	}
+
+	/**
+	 * Returns human-friendly help for the generator.  Provide the consumer with help
+	 * tips, parameters here
+	 *
+	 * @return A string value for the help message
+	 */
+	public String getHelp() {
+		return "Generates a typescript client library.";
+	}
+
+	public TypescriptGenerator() {
+		super();
+		outputFolder = "build/typescript";
+		modelTemplateFiles.put(
+			"types/type.mustache",
+			".ts"
+		);
+
+		// We want the package lock files to be detectable and maintained by dependabot.
+		// This means they cannot be templates
```

Comment:
I should put this comment closer to the lines they refer to (`package.json` and `package-lock.json`)